pkg/proxy/iptables/proxier.go

```diff
@@ -1183,6 +1186,22 @@ func (proxier *Proxier) syncProxyRules() {
 				// Jump to the external destination chain.  For better or for
 				// worse, nodeports are not subect to loadBalancerSourceRanges,
 				// and we can't change that.
+				if proxier.localhostNodePorts && proxier.ipFamily == v1.IPv4Protocol {
+					rule := []string{
+						"-A", string(kubeNodePortsChain),
+						"-m", "comment", "--comment", svcPortNameString,
+						"-m", protocol, "-p", protocol,
+						"-d", "127.0.0.0/8",
+						"--dport", strconv.Itoa(svcInfo.NodePort()),
+					}
+					if proxier.nfAcctCounters[metrics.LocalhostNodePortAcceptedNFAcctCounter] {
```

Comment:
you should merge this check in with the `if` above; we only need a separate 127.0.0.0/8 rule if the nfacct counter exists